src/coll/mod.rs

```diff
@@ -184,35 +196,32 @@ impl Collection {
                               options: Option<FindOptions>,
                               cmd_type: CommandType)
                               -> Result<Cursor> {
-        let options = options.unwrap_or_else(FindOptions::new);
-        let flags = OpQueryFlags::with_find_options(&options);
-
-        let doc = if options.sort.is_some() {
-            let mut doc = bson::Document::new();
-            doc.insert(String::from("$query"),
-                       Bson::Document(filter.unwrap_or_else(bson::Document::new)));
-
-            doc.insert(String::from("$orderby"),
-                       Bson::Document(options.sort.as_ref().unwrap().clone()));
-
-            doc
-        } else {
-            filter.unwrap_or_else(bson::Document::new)
+        let find_options = options.unwrap_or_else(FindOptions::new);
+        let flags = OpQueryFlags::with_find_options(&find_options);
+
+        let doc = match find_options.sort {
+            Some(ref sort_opt) => {
+                doc! {
+                    "$query" => (filter.unwrap_or_else(bson::Document::new)),
+                    "$orderby" => (Bson::Document(sort_opt.clone()))
+                }
+            }
+            None => filter.unwrap_or_else(bson::Document::new),
         };
 
-        let read_pref = options.read_preference.unwrap_or(self.read_preference.to_owned());
+        let read_preference = match find_options.read_preference {
+            Some(ref read_preference_option) => read_preference_option.clone(),
+            None => self.read_preference.clone(), 
+        };
 
         Cursor::query(self.db.client.clone(),
                       self.namespace.to_owned(),
-                      options.batch_size,
                       flags,
-                      options.skip as i32,
-                      options.limit,
                       doc,
-                      options.projection.clone(),
+                      find_options,
                       cmd_type,
                       false,
-                      read_pref)
+                      read_preference)
```

Comment:
Interesting. I wonder if we can remove this read_pref option here and skip the mandatory cloning? Perhaps something to look into at a later point.